js/src/components/AnalysisCheck.js

```diff
@@ -0,0 +1,30 @@
+import { SvgIcon } from "@yoast/components";
+import { getIconForScore } from "./contentAnalysis/mapResults";
+import * as PropTypes from "prop-types";
+
+/**
+ * Renders a pre-publish score in the from of an icon, label
+ * and the score value written out.
+ *
+ * @param {"good"|"ok"|"bad"|"loading"} score The score, as a string.
```

Comment:
can we use BlockValidation here? if all blocks are BlockValidation.Valid, you have a perfect score.
I'm adding a getValidationResult selector in P2-364 which could be used here as well, to get the validation result for a given clientId from the store.

Reply:
representations shouldn't be used in logic anywhere but in the presenter; up to here all logic is based on BlockValidation

Reply:
This component is also used for the other results in the sidebar, e.g. the SEO analysis result and the readability analysis result.

Refactoring this (for example to use an enum) is possible but not advisable, since other code may depend on these strings.